core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -562,6 +578,28 @@ public synchronized WorkerNetAddress getLocalWorker() throws IOException {
     return mLocalWorker;
   }
 
+  /**
+   * @return the info of all block workers eligible for reads and writes
+   */
+  public synchronized List<BlockWorkerInfo> getCachedWorkers() throws IOException {
+    if (mWorkerInfoList == null || mWorkerRefreshPolicy.attempt()) {
+      mWorkerInfoList = getAllWorkers();
+    }
+    return mWorkerInfoList;
+  }
+
+  /**
+   * @return the info of all block workers
```

Comment:
can you indicate that this method is more expensive than `getCachedWorkers`?

Reply:
Added thanks